File: benchmarks/python/11_object_property_loop.py
class Point5D:
    __slots__ = ('x', 'y', 'z', 'w', 'v')
    def __init__(self, x, y, z, w, v):
        self.x = x
        self.y = y
        self.z = z
        self.w = w
        self.v = v

def object_property_loop(n):
    p = Point5D(1.0, 2.0, 3.0, 4.0, 5.0)
    s = 0.0
    i = 0
    while i < n:
        s += p.x + p.y + p.z + p.w + p.v
        i += 1
    return s
print(object_property_loop(10000000))

File: benchmarks/node/11_object_property_loop.mjs
function object_property_loop(n) {
    const p = { x: 1.0, y: 2.0, z: 3.0, w: 4.0, v: 5.0 };
    let sum = 0;
    for (let i = 0; i < n; i++) {
        sum += p.x + p.y + p.z + p.w + p.v;
    }
    return sum;
}
console.log(object_property_loop(10000000));
